### PR TITLE
Add Smithsonian harvester

### PR DESCRIPTION
**Pull request by marktriggs**
Here's an implementation of a parser for the Smithsonian records.

Since this is a file-based harvest, I've set up the harvester so that the `:uri` parameter is taken as a local directory (similar to how we had the Hathi harvester treat `:uri` as a filesystem path).  Any .xml or .xml.gz files in that directory get harvested.  Originally we had it just harvesting one file at a time, but that seems inconvenient if we're talking about 20-30 files.

I think that's the only remarkable thing about this one, but happy to incorporate any feedback.

Thanks!
Mark

### CODE REVIEW

File: app/harvesters/smithsonian_harvester.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
+##
+# A harvester implementation that can handle a directory of Smithsonian
+# .xml[.gz] files.
+#
+# A typical invocation is:
+#
+#   SmithsonianHarvester.new(uri: '/data-files/Smithsonian/')
+#
+# Note that `:uri` is a local directory in this case.  Any files ending in
+# '.xml.gz' (case insensitive) will be harvested.
```
Comment (anarchivist) [inline app/harvesters/smithsonian_harvester.rb]: Should this be `'.xml' or '.xml.gz'`?

Comment (marktriggs) [inline app/harvesters/smithsonian_harvester.rb]: Yup!  Will fix

**Comment by anarchivist** [comment]
Looks good to me; assigning to @markbreedlove for a second glance.

**Comment by moltude** [comment]
looks good to me

**Comment by marktriggs** [comment]
Thanks all :)  I've just fixed the comment that @anarchivist identified too.

**Comment by anarchivist** [comment]
Thanks @marktriggs!

**Pull request merged**